framework/db/Command.php

```diff
@@ -829,30 +829,32 @@ public function execute()
     {
         $sql = $this->getSql();
 
-        $rawSql = $this->getRawSql();
-
-        Yii::info($rawSql, __METHOD__);
+        $log = $this->db->enableQueryLog;
+        if ($log) {
+            $rawSql = $this->getRawSql();
+            Yii::info($rawSql, __METHOD__);
+            $token = $rawSql;
+        }
 
         if ($sql == '') {
             return 0;
         }
 
         $this->prepare(false);
 
-        $token = $rawSql;
         try {
-            Yii::beginProfile($token, __METHOD__);
+            $log && Yii::beginProfile($token, __METHOD__);
```

Comment:
I would better introduce separeted methods for log and profile composition like I did at `yii2-mongodb`:
https://github.com/yiisoft/yii2-mongodb/blob/master/Command.php#L179-L195

It will look better then `&&` operator.